pkg/cmd/pulumi/destroy.go

```diff
@@ -201,7 +201,7 @@ func newDestroyCmd() *cobra.Command {
 				sm = snap.SecretsManager
 			}
 
-			cfg, sm, err := getStackConfiguration(ctx, s, proj, sm)
+			cfg, sm, err := getStackConfiguration(ctx, s, proj, &sm)
```

Comment:
This reads like non-obvious behavior. 
We're passing a ref to `sm`, but we're also overwriting `sm` in the same statement.
Dropping the pointer for the interface will fix the issue.